docs: update Dynamic Component loader docs to rely on `ngComponentOutlet` (#49915)
angular/angular: aio/content/guide/dynamic-component-loader.md

<a id="directive"></a>

> ## Rendering components dynamically

Reviewer: It looks like we have 2 sections with the same name. Should we just merge them together?
We can also drop the "Angular comes with its own API for loading components dynamically." sentence.

The `NgComponentOutlet` directive can be used to instantiate components and insert them into the current view. This directive allows you to provide a component class that should be rendered, as well as component inputs to be used during initialization.

<code-example header="src/app/ad-banner.component.ts" path="dynamic-component-loader/src/app/ad-banner.component.ts" region="component"></code-example>

The `AdBannerComponent` class injects the `AdService` service and requests a list of ads. 
The "current ad" index is set to `0` initially to indicate that the first ad should be displayed. 
When a user clicks the "Next" button, the index is increased by one. 
Once the index reaches the length of the ads array, the index is reset back to `0`.

In the template, the `currentAd` getter is used to retrieve a current ad. 
If the value changes, Angular picks it up and reflects the changes in the UI.

## Different components from the service

> In the ad banner, all components don't need to be related as we can pass seamlessly the inputs to the `ngComponentOutlet`. 

Reviewer: ```suggestion
Components returned from the `AdService` service and used in `NgComponentOutlet` in the `AdBannerComponent` template can be different. Angular detects if a component class has changed and updates the UI accordingly.
```

<a id="directive"></a>

## Rendering components

> The `NgComponentOutlet` directive is used to instantiates the components and inserts them in the current View. 

Reviewer: ```suggestion
The `NgComponentOutlet` directive is used to instantiate components and insert them into the current view. 
```

It accepts as arguments, among others, the component type and its inputs. 

<code-example header="src/app/ad-banner.component.ts (template)" path="dynamic-component-loader/src/app/ad-banner.component.ts" region="ng-container"></code-example>

> ## Selecting a component to render

Reviewer: It looks like some of the information in this section and the section above is outdated (refer to an old implementation). I'd propose merging both sections into one and displaying an entire code snippet (do **not** split a template and the class itself).

The content of the "merged" section itself can look like this:

```
## Rendering components dynamically

The `NgComponentOutlet` directive can be used to instantiate components and insert them into the current view. This directive allows you to provide a component class that should be rendered, as well as component inputs to be used during initialization.

<<< Complete code of `AdBannerComponent` class here >>>

The `AdBannerComponent` class injects the `AdService` service and requests a list of ads. The "current ad" index is set to `0` initially to indicate that the first ad should be displayed. When a user clicks the "Next" button, the index is increased by one. Once the index reaches the length of the ads array, the index is reset back to `0`.

In the template, the `currentAd` getter is used to retrieve a current ad. If the value changes, Angular picks it up and reflects the changes in the UI.
```

Reviewer: We could also tweak the code a bit to make it a bit more straightforward and to match the content:

```
  displayNextAd() {
    this.currentAdIndex++;
    // Reset the current ad index back to `0` when we reach the end of an array.
    if (this.currentAdIndex === this.adList.length) {
      this.currentAdIndex = 0;
    }
  }
```

> ## Resolving components

Reviewer: ```suggestion
## Selecting a component to render
```
